binary/capabilities.proto

```diff
@@ -0,0 +1,26 @@
+syntax = "proto3";
+
+package io.substrait;
+
+option java_multiple_files = true;
+option csharp_namespace = "Substrait.Protobuf";
+
+// Defines a set of Capabilities that a system (producer or consumer) supports.
+message Capabilities {
+
+    // List of Substrait versions this system supports
+    repeated string substrait_versions = 1;
+
+    // list of com.google.Any message types this system supports for advanced extensions.
+    repeated string advanced_extension_type_urls = 2;
+
+    // list of simple extensions this system supports.
+    repeated SimpleExtension simple_extensions = 3;
+
+    message SimpleExtension {
+        string uri = 1;
+        repeated string function_keys = 2;
+        repeated string type_keys = 3;
+        repeated string type_variation_keys = 4;
```

Comment:
I think these need documentation, and possibly examples.

Reply:
agreed. baby steps, baby!